### PR TITLE
Reduce the memory footprint

### LINKED ISSUE #3: `store()` method does not stream data to GCS, uses byte[].
I just encountered an OOM / Java heap space error in a gradle build and thought i should check the implementation, in the `store()` method in the service it appears that the output to be cached is being buffered in a ByteArrayOutputStream before being passed to GCS SDK.

https://github.com/tehlers/gradle-gcs-build-cache/blob/4f6e4bea7f8b1f0d7e70b86ba1896268c5983c45/src/main/kotlin/net/idlestate/gradle/caching/GCSBuildCacheService.kt#L63

For tasks with large outputs (mine is 100mb+) this means the entire thing is going to be held in memory, I can't see any reason why you could just pass an InputStream  to GCS SDK instead 👍

### PR DESCRIPTION
**Pull request by KengoTODA**
Thanks for making this plugin shared and open!
I am using this plugin and it is very useful, so I want to suggest a PR to improve its execution performance.

This PR will close #3 by replacing `ByteArrayOutputStream` with [`FileBackedOutputStream` provided by Guava](https://guava.dev/releases/30.1.1-jre/api/docs/com/google/common/io/FileBackedOutputStream.html). Guava is introduced by the dependency on `google-cloud-storage`, so it's already in the CLASSPATH of this plugin.

```
\--- com.google.cloud:google-cloud-storage:1.118.1
     +--- com.google.guava:guava:30.1.1-jre
```

I also updated the Gradle used by this repository, with [`gradle/wrapper-validation-action`](https://github.com/gradle/wrapper-validation-action) that proves the changed wrapper is valid and not vulnerale.

### CODE REVIEW

**Comment by KengoTODA** [comment]
About the Workflow, you can refer [the workflow run that run in my repository](https://github.com/KengoTODA/gradle-gcs-build-cache/runs/7871880221?check_suite_focus=true) as example.

File: src/main/kotlin/net/idlestate/gradle/caching/GCSBuildCacheService.kt [hunk 1]
```diff
@@ -104,4 +106,11 @@ class GCSBuildCacheService(credentials: String, val bucketName: String, val refr
     override fun close() {
         // nothing to do
     }
+
+    companion object {
+        /**
+         * The threshold to write data onto storage.
+         */
+        const val FILE_THRESHOLD = 8 * 1024 * 1024
```
Comment (KengoTODA) [inline src/main/kotlin/net/idlestate/gradle/caching/GCSBuildCacheService.kt:114]: Honestry speaking, there is no strong reason to pick `8MiB` as threshold.

Comment (tehlers) [reply src/main/kotlin/net/idlestate/gradle/caching/GCSBuildCacheService.kt:114]: I will probably add a configuration option for the threshold.

**Comment by tehlers** [comment]
Thanks!

**Pull request merged**